File: Android/app/src/main/java/com/pnu/aidbtdiary/entity/DbtDiary.kt
package com.pnu.aidbtdiary.entity

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.time.LocalDate
import java.time.LocalDateTime

@Entity(tableName = "dbt_diary")
data class DbtDiary (
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
     val date: LocalDate,
     val situation: String,
     val emotion: String,
     val intensity: Int,
     val thought: String,
     val behavior: String,
     val dbtSkill: String,
    val solution: String = "",
    val deleted: Boolean = false,
    val sentiment: Boolean,
    val createdAt: LocalDateTime,
    val updatedAt: LocalDateTime
)

File: BE/app/models/dbt_diary.py
from datetime import date, datetime
from pydantic import BaseModel

class DbtDiaryBase(BaseModel):
    date: date
    situation: str
    emotion: str
    intensity: int
    thought: str
    behavior: str
    dbt_skill: str
    sentiment: bool
    is_synced: bool = False
    created_at: datetime
    updated_at: datetime


class DbtDiaryCreate(DbtDiaryBase):
    pass


class DbtDiary(DbtDiaryBase):
    id: int

    class Config:
        from_attributes = True